File: AlbumsToListenToAtWork/AlbumsToListenToAtWork.py
# Author : Herve Nyemeck
# Copy lines from google doc
# Randomly pick a line from pre-existing file
# Output said line and send to user

#Album_List = open("Albums", "r") 
# print (Album_List)
import sendEmail 
from random import randint
import smtplib
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText
from email.mime.base import MIMEBase
from datetime import datetime
import os
import sys

albumListFile = "Albums to Listen to at Work.txt"
oldAlbumsListFile = "OldAlbums.txt"
today = "Today.txt"

def removePreexisting():
    exists = os.path.isfile(today)
    if exists:
        os.remove(today)
    else:
        return  

removePreexisting()

file = open(albumListFile,"r")
lines = []
for line in file:
    lines.append(line)
size = len(lines)

#if lines = []:                     Add statement for if albums are full/exist in old albums file
#    print("Out of Albums")
# else:

album=""
def generateAlbum():    
    random_number = randint(0,(size-1))
    global album
    album = lines[random_number]
generateAlbum()
file.close()

old_albums = open(oldAlbumsListFile,"r")
old_albums_list = []

for line in old_albums:
    old_albums_list.append(line)

#def runItBack():
while album in old_albums_list:
    generateAlbum()

old_albums.close()
old_albums = open(oldAlbumsListFile,"a")
old_albums.write(album)
# print(album)
file.close()
today_album = open(today, "w")
today_album.write(album)
today_album.close()
sendEmail.sendEmail()
sys.exit()
#print (file.readlines())

# msg = EmailMessage()
#runItBack()

File: AlbumsToListenToAtWork/AlbumsToListenToAtWork.java
/*
Author: Nicolas Rotondaro
*/

import java.io.IOException;
import java.nio.file.*;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) {
        //Creating imaginary object to call methods with
        Main main = new Main();

        //Methods to run the script()
        main.removePreexisting();
        main.generateAlbum();
        main.runItBack();
    }

     //Converting text files into lists for java purposes
     public ArrayList generateLists(File fileName) {
        List generatedList = new ArrayList<>();

        try (BufferReader br = new BufferedReader(new FileReader(fileName))) {
            String currentLine;

            while ((currentLine = br.readLine()) != null) {
                generatedList.add(currentLine);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        return generatedList;
    }

    //Deleting old files first
    public void removePreexisting() {
        if (todayList.exists()) {
            todayList.delete();
        } else {}
    }

    //Should replace file names with file location on local computer
    //Creating lists to be able to edit/write
    public List albumList = generateLists("Albums to Listen to at Work.txt");
    public List oldAlbumsList = generateLists("OldAlbums.txt");
    public List todayList = generateLists("Today.txt");

    //Random Number Generator
    public int getRandomNumber() {
        double randomNum = Math.random();
        randomNum = randomNum * albumList.size();
        int randomInt = (int) randomNum;
        return randomInt;
    }

    //Picking random album from list
    public String generateAlbum() {
        album = "";

        //Need to put filepath instead of name
        try (Stream<String> albumLines = Files.lines(Paths.get("Albums to Listen to at Work.txt"))) {
            album = albumLines.skip(getRandomNumber()).findFirst().get();
        }

        return album;
    }

    //Checking if album is present in oldAlbum list
    public void runItBack() {
        boolean oldSong = oldAlbumList.contains(album);

        //If present, find new album and try again
        if (oldSong == true) {
            generateAlbum();
            runItBack();
        } 
        //If not, add to todayList and oldAlbumsList
        else {
            oldAlbumsList.add(album);
            todayList.add(album);
        }
    }
}
